src/sparcscore/ml/plmodels.py

```diff
@@ -192,6 +192,9 @@ def configure_optimizers(self):
                 self.hparams["weight_decay"] = 0
             optimizer = torch.optim.Adam(self.parameters(), lr=self.hparams["learning_rate"], weight_decay=self.hparams["weight_decay"])
         elif self.hparams["optimizer"] == "AdamW":
+            #set weight decay to 1e-2 if not specified in hparams 
+            if self.hparams["weight_decay"] is None:
+                self.hparams["weight_decay"] = 10 ** -2
```

Comment:
I think it's good to add it. then we ensure that we write out all required hparams